src/pages/ShowFeelingsPage/ShowFeelingTemplate.tsx

```diff
@@ -1,59 +1,285 @@
 /* eslint-disable react/prop-types */
-import React from 'react';
+import React, { useState } from 'react';
 import {
-  Button, Nav, Container,
+  Button, Nav, Container, Modal,
 } from 'react-bootstrap';
 import { useRecoilValue } from 'recoil';
 import { useTranslation } from 'react-i18next';
+import {
+  ChevronDown, ChevronRight, TrashFill, Journal,
+} from 'react-bootstrap-icons';
 
 import { darkModeState } from '@store';
+import { IFeelingItem } from '@models';
+import { removeFeeling, addFeelingNote, removeFeelingNote } from '@api/FeelingsAPI';
+import { feelingListType } from '@src/global';
+import { feelingsEmojis } from '@consts/FeelingsList';
 
 import '@translations/i18n';
 import './ShowFeelingsPage.scss';
 
-interface IShowFeelingTemplateProp {
-  id: number;
-  category: string;
-  content: string;
-  date: Date;
+interface ISetFeelingsListObject {
+  feelingsList: feelingListType[],
+  setFeelingsList: React.Dispatch<React.SetStateAction<feelingListType[]>>
+}
+
+interface IProps {
+  feelingsListObject: IFeelingItem[],
+  setFeelingsListObject: ISetFeelingsListObject,
+  currentFeelingsList: feelingListType[],
 }
 
-export const ShowFeelingTemplate: React.FC<IShowFeelingTemplateProp> = ({ feelingsListObject }) => {
+export const ShowFeelingTemplate: React.FC<IProps> = ({
+  feelingsListObject,
+  setFeelingsListObject,
+  currentFeelingsList,
+}) => {
   const { t } = useTranslation();
   const darkModeStatus = useRecoilValue(darkModeState);
+  const [showInputModal, setShowInputModal] = useState(false);
+  const [showNotesModal, setShowNotesModal] = useState(false);
+  const [showOptions, setShowOptions] = useState(false);
+  const [selectedFeeling, setSelectedFeeling] = useState<number>();
+  const [selectedFeelingNote, setSelectedFeelingNote] = useState('');
+  const [noteValue, setNoteValue] = useState('');
+
+  const handleInputClose = () => setShowInputModal(false);
+  const handleInputShow = () => setShowInputModal(true);
+  const handleNotesClose = () => setShowNotesModal(false);
+  const handleNotesShow = () => setShowNotesModal(true);
+  const handleFeelingsNoteModify = async () => {
+    addFeelingNote(selectedFeeling!, noteValue)
+      .then((newFeelingsList) => {
+        // @ts-ignore
```

Comment:
https://github.com/Microsoft/TypeScript/issues/24587 Let me know if you find a way to resolve this error `Type 'Date' cannot be used as an index type`

Reply:
Not working, creating additional errors too. The logic seems fine and since there is no extension at the moment the priority can be shifted to release rather than fix these problems. I will create an issue for this but let's get this into production for now.